README.md

```diff
@@ -1,3 +1,262 @@
+
+There are plans to enable safer renaming if you intend for your TypeScript code to be used as an external library (in the case that you have a public API). At the moment, however, there is no support for this.
+
+###Caveats/Warnings
+
+---
+In order to get the most out of the minifier, it is recommended that the user types their program as specifically and thoroughly as possible. In general, you should  avoid using the `any` type, and implicit `any`s in your code as this might result in unwanted naming.  
+
+We recommend that you write TypeScript with the  “no implicit any” compiler flag enabled, and try transpiling your code with the TS compiler before trying the minifier.
+
+Please avoid explicit any casting: 
+
+```javascript
+// example of explicit any casting
+var x = 7; 
+<any>x;
+```
+
+###Preliminary Measurements
+
+---
+TS-Minify was run on a well-typed TypeScript program with the following stats and results:
+
+- Approximately 2100 lines of code (split across 10 files)
+- Unminified and Uglified: 72kb
+- Minified and Uglified: 56kb (codesize reduction of about 20%)
+- About 6% - 8% codesize reduction after minification, Uglification, and GZIP compression.
+
+###Contributing
+
+---
+Clone the repository from GitHub:
+
+```shell
+$ git clone git@github.com:angular/ts-minify.git
+$ cd ts-minify
+$ npm install
+```
+
+Run the unit tests with `gulp unit.test`
+
+Run the end-to-end tests with `gulp e2e.test`.
+
+This project uses `clang-format`. Run `gulp test.check-format` to make sure code you write conforms to this standard.
+
+- If you need some guidance on understanding TypeScript, look at the [TypeScript GitHub Wiki](https://github.com/Microsoft/TypeScript/wiki).
+- If you need a quick introduction to the TS Compiler API, take a look at the page on using the [TS Compiler](https://github.com/Microsoft/TypeScript/wiki/Using-the-Compiler-API) API. 
+- Take a look at the `typescript.d.ts` for type signatures, and to understand what is available to you from the TS toolchain.
+
+###Contributors
+
+---
+Daria Jung (lowly Google intern)
+
+
+###License: Apache 2.0
+
+---
+```
+Copyright 2015 Google, Inc. http://angular.io
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+```
+###Future
+
+---
+
+There are a lot of [issues](https://github.com/angular/ts-minify/issues)! Property renaming is hard! 
```

Comment:
Don't scare people off :) Can you identify some low-hanging fruit ones, so that we can attract community contributions.